otsdaq/CodeEditor/CodeEditor.cc

```
@@ -513,21 +515,40 @@ void CodeEditor::getFileGitURL(cgicc::Cgicc& cgiIn, HttpXmlDocument* xmlOut)
 std::string CodeEditor::getFileGitURL(const std::string& basepath,
                                       const std::string& path)
 {
-	//TODO! October 2024 by rrivera
-	//TODO! request with linux exec to spack environment
-
 	__COUTV__(basepath);
-	__COUTV__(path);
-	std::string fullpath;
-	if(path.find(basepath) == 0)  //check if path is already complete
-		fullpath = path;
-	else
-		fullpath = basepath + "/" + path;
-	__COUTV__(fullpath);
+	std::string localPath = path;
+
+	std::string package = localPath.substr(0, localPath.find("/"));
+
+	std::string cmd  = "spack info " + package;
+	FILE*       pipe = popen(cmd.c_str(), "r");
+	if(!pipe)
+		return "";
+
+	std::stringstream result;
+	char              buffer[256];
+	while(fgets(buffer, sizeof(buffer), pipe) != nullptr)
+		result << buffer;
+	pclose(pipe);
+
+	std::string outStr = result.str();
+
+	std::regex  urlRegex(R"(https:\/\/github\.com\/[^\s]+)");
+	std::smatch urlMatch;
+	if(!std::regex_search(outStr, urlMatch, urlRegex))
+		return "";
```

Is this an error case? How does this error get propagated to web user? 

Likely you should throw an exception for these errors using { __SS__ << "Error detail here"; __SS_THROW__;}

---

throwing an exception like that will propagate error back to user web page